langlib/lang.query/src/main/ballerina/helpers.bal

```diff
@@ -168,52 +178,85 @@ function addToTable(stream<Type, CompletionType> strm, table<map<Type>> tbl, err
         // and make it immutable with createImmutableTable().
         table<map<Type>> tempTbl = table [];
         table<map<Type>> tbl2 = createTableWithKeySpecifier(tbl, typeof(tempTbl));
-        table<map<Type>> tempTable = check createTable(strm, tbl2, err);
+        table<map<Type>> tempTable = check createTable(strm, tbl2);
         return createImmutableTable(tbl, tempTable.toArray());
     }
-    return createTable(strm, tbl, err);
+    return createTable(strm, tbl);
 }
 
-function createTable(stream<Type, CompletionType> strm, table<map<Type>> tbl, error? err) returns table<map<Type>>|error {
+function createTable(stream<Type, CompletionType> strm, table<map<Type>> tbl) returns table<map<Type>>|error {
     record {| Type value; |}|CompletionType v = strm.next();
     while (v is record {| Type value; |}) {
         error? e = trap tbl.add(<map<Type>> checkpanic v.value);
         if (e is error) {
-            if (err is error) {
-                return err;
-            }
             tbl.put(<map<Type>> checkpanic v.value);
         }
         v = strm.next();
     }
+    if v is error {
+        return v;
+    }
+    return tbl;
+}
+
+function addToTableForOnConflict(stream<Type, CompletionType> strm, table<map<Type>> tbl, boolean isReadOnly) 
+    returns table<map<Type>>|error {
+    if isReadOnly {
+        // TODO: Properly fix readonly scenario - Issue lang/#36721
+        // In this case tbl will be an immutable table. Therefore, we will create a new mutable table. Next, we will
+        // pass the newly created table into createTableWithKeySpecifier() to add the key specifier details from the
+        // original table variable (tbl). Then the newly created table variable will be populated using createTable()
+        // and make it immutable with createImmutableTable().
+        table<map<Type>> tempTbl = table [];
+        table<map<Type>> tbl2 = createTableWithKeySpecifier(tbl, typeof(tempTbl));
+        table<map<Type>> tempTable = check createTableForOnConflict(strm, tbl2);
+        return createImmutableTable(tbl, tempTable.toArray());
+    }
+    return createTableForOnConflict(strm, tbl);
+}
+
+function createTableForOnConflict(stream<Type, CompletionType> strm, table<map<Type>> tbl) 
+    returns table<map<Type>>|error {
+    record {| Type value; |}|CompletionType v = strm.next();
+    while (v is record {| Type value; |}) {
+        record {|Type v; error? err;|}|error value = trap (<record {|Type v; error? err;|}> checkpanic v.value);
```

Comment:
We don't use checkpanic as practrics?. Is this a special case?

Reply:
This the pattern we have followed so far. We may have to send another PR refactoring these.